doc/rst/tools/mason/guide/prereqs.rst

```diff
@@ -0,0 +1,130 @@
+:title: Mason Guide: *Building Code in Other Languages*
+
+Building Code in Other Languages
+================================
+
+Some Mason packages may include code in languages other than Chapel. A common
+scenario for this is when the package needs to interoperate with other
+languages. The code in other languages in such a scenario is called
+"prerequisites". Chapel 2.7 release includes Mason with initial support for
+prerequisites.
+
+Understanding Prerequisites
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+While they are similar to dependencies, prerequisites are inherent part of the
+Mason package whereas dependencies need to be built and installed before Mason
+can start building a package. Consider a scenario in which your Mason package
+has a dependency on a C or C++ library, and that you need to implement a shim
+between that library and the Chapel code. That shim needs to be compiled before
+the Chapel code and needs to be linked. Obviously, the Chapel compiler cannot be
+used in this case. Moreover, depending on the library, build process can be
+completely different from one scenario to another.
+
+Mason can use other tools to build such code that is part of your package.
+
+.. warning::
+
+   As of Chapel 2.7, prerequisite support is prototypical. We plan to expand
+   the capabilities and make the implementation more robust in future releases.
+   If you encounter any issue, or have a feature request related to
+   prerequisites, please `create an issue <https://github.com/chapel-lang/chapel/issues/new?template=01-feature_request.md>`_.
+
+
+
+A "Hello, world!" Example
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Consider a Mason application where "Hello, world!" is printed through a C
+function that Chapel calls. The key source files in this scenario are:
+
+
+* Hello.chpl
+
+.. code-block:: chapel
+
+   module prereq1 {
+     extern proc hello();
+
+     proc main() {
+       hello();
+     }
+   }
+
+* hello.h
+
+.. code-block:: c
+ 
+   void hello(void);
+
+* hello.c
+
+.. code-block:: c
+ 
+   #include <stdio.h>
+
+   #include "hello.h"
+  
+   void hello(void) {
+     printf("Hello world!\n");
+   }
+
+As of Chapel 2.7, building this project with Mason requires:
+
+* A ``prereqs`` directory at the project root,
+* Each prerequisite to have a subdirectory within ``prereqs``,
+* Within each prerequisite's directory a ``Makefile``,
+* Each ``Makefile`` to support two targets
+    * The default ``make``: To build the external code into something that Chapel
+      compiler link, typically a object file
+    * ``make printchplflags``: To print flags to be passed to ``chpl`` for this
+      prerequisite to be linked with the Chapel application or library.
+
+Based on these requirements, the file structure should look like:
```

Comment:
i could be wrong. but i think for this to render correctly it needs spaces between each bullet